examples/dev/interaction_box.py

```diff
@@ -0,0 +1,85 @@
+import warnings
+
+import numpy as np
+from magicgui import magicgui
+
+import napari
+from napari._vispy.overlays.interaction_box import VispySelectionBoxOverlay
+from napari._vispy.utils.visual import overlay_to_visual
+from napari.components.overlays import SelectionBoxOverlay
+from napari.layers import Image
+
+
+# define a model for the selection box overlay;
+# we subclass from SelectionBoxOverlay to get the
+# default behavior of the selection box;
+class SelectionBoxNoRotation(SelectionBoxOverlay):
+    """Selection box overlay with no rotation handle."""
+
+
+# we also need to define an equivalent vispy overlay;
+# again, we subclass from VispySelectionBoxOverlay
+class VispySelectionBoxNoRotation(VispySelectionBoxOverlay):
+    """Vispy selection box overlay with no rotation handle."""
+
+    # the _on_bounds_change method is the same as in the
+    # original VispySelectionBoxOverlay, but we set
+    # rotation to False to not draw the rotation handle
+    def _on_bounds_change(self):
+        if self.layer._slice_input.ndisplay == 2:
+            top_left, bot_right = self.overlay.bounds
+            self.node.set_data(
+                # invert axes for vispy
+                top_left[::-1],
+                bot_right[::-1],
+                handles=self.overlay.handles,
+                selected=self.overlay.selected_handle,
+                # by setting rotation to False,
+                # the circle handle will not be drawn
+                rotation=False,
+            )
+
+
+# before we can use the new overlay, we have to update
+# the overlay_to_visual mapping to include our new overlay;
+# this is necessary so that the correct vispy overlay
+# is used when the overlay is created
+overlay_to_visual[SelectionBoxNoRotation] = VispySelectionBoxNoRotation
+
+viewer = napari.Viewer()
+
+# we add an image layer with random data;
+data = np.random.randint(0, 255, size=(1024, 1024), dtype=np.uint8)
+image = viewer.add_image(
+    data,
+    name='image',
+)
+
+# just for type checking
+assert isinstance(image, Image)
+
+# we recover the bounds of the image layer;
+# this method will ensure that the overlay is drawn
+# correctly in the viewer
+layer_bounds = tuple(tuple(x.tolist()) for x in image._display_bounding_box_augmented([0, 1]).T)
```

Comment:
I must admit I'm not a great fan of this because it feels a bit too convoluted to recover the bounds. Probably a dedicated `visual_bounds` property could be provided in the future?

Reply:
I'm not sure why the validator doesn't handle converting array to tuples... but otherwise this is as intended. In general, the "visual bounds" are not a thing that the user should care about, unlike the data extents.

In fact, this stuff might be better moved to the vispy module in the future, as this is vispy specific and might differ with other backends.